File: backend/warp-webservice/src/product/product.rs
use serde::{Serialize,Deserialize};


pub type Id = u32;

#[derive(Serialize, Deserialize, Debug)]
#[serde(rename_all = "camelCase")]
pub struct Product {
	pub product_id: Id,
	pub manufacturer: String,
	pub sku: String,
	pub upc: String,
	pub price_per_unit: String,
	pub quantity_on_hand: usize,
	pub product_name: String,
}

File: frontend/src/model/product.ts

export class Product {
    productId: string;
    manufacturer: string;
    sku: string;
    upc: string;
    pricePerUnit: string;
    quantityOnHand: string;
    productName: string;

    constructor(
        productId: string,
        manufacturer: string,
        sku: string,
        upc: string,
        pricePerUnit: string,
        quantityOnHand: string,
        productName: string,
    ) {
        this.productId = productId;
        this.manufacturer = manufacturer;
        this.sku = sku;
        this.upc = upc;
        this.pricePerUnit = pricePerUnit;
        this.quantityOnHand = quantityOnHand;
        this.productName = productName;
    }
}
